client/src/components/Signup.jsx

```diff
@@ -0,0 +1,170 @@
+import React, { Component } from 'react';
+
+import { Link, Redirect } from 'react-router-dom';
+import { Button, Card, Form, Header, Icon, Input } from 'semantic-ui-react';
+import isEmail from 'validator/lib/isEmail';
```

Comment:
Good job but don't you think it is better to place the validator in a separate file for separation of concern.